### ISSUE TITLE
render file: broken?

### ISSUE THREAD

**Issue by eiked** [rails/rails#24076]
using Rails 4.2.5.2

When I do:
    render(file:'/tmp/foo.jpg')

I get the exception:
    Missing template tmp/foo.jpg

(please note that the leading slash is removed)

Expected result:
   send_file('/tmp/foo.jpg') should be called

Actual result:
    Instead of sending the file, a template is searched

Comments:

I have not tried this api before, so maybe I just got it wrong.

My expectation from the documentation at:
- http://guides.rubyonrails.org/layouts_and_rendering.html#using-render
was, that render(file:'foo') should invoke send_file('foo'):
- http://api.rubyonrails.org/classes/ActionController/DataStreaming.html#method-i-send_file

**Comment by rafaelfranca** [comment]
Thank you for the reporting. Just to understand better where did you saw in the linked documentation that `render file:` should invoke `send_file`?

**Comment by rafaelfranca** [comment]
I just checked the behavior and `render file: '/tmp/foo.txt'` is working as expected.

I got `Missing template tmp/foo.txt` when I called `render '/tmp/foo.txt'`. This is the expected behavior in 4.2.5.2. This was exactly the security fix released in the 4.2.5.1 version. The guides needs to be changed to match the new behavior.

As per `render file:` calling `send_file` that was never the expected behavior.

**Comment by rafaelfranca** [comment]
Fixed the guides in 403c57a

**Comment by eiked** [comment]
Hi rafaelfranka,
got it fixed.

now using: send_file() (instead of render(file:))

Looks like I got the documentation wrong.
The power is yours!
https://www.youtube.com/watch?v=7V7M6LxzcFc

**Comment by rafaelfranca** [comment]
Great! Thank you for reporting and confirming the fix. The documentation was improved with your help!

**Comment by eiked** [comment]
but it still does not work for me,
I now believe it must be a problem in the passenger,
because it worked for me locally,
and now reimplementing that in a completely different way,
it again worked for me locally like a charme.
(in dev and production)
but then again it fails again (no: takes 4 minutes to render)
when I set this up in my passenger setup.
Looks like there is a problem still lurking somewhere.

I keep on debugging this. I'll let you know.

oops? we listed some kiss at rockers de
